ecllibrary/teststd/DataPatterns/TestBenford.ecl

```diff
@@ -0,0 +1,168 @@
+IMPORT Std.DataPatterns;
+
+EXPORT TestBenford := MODULE
+
+    SHARED TestData := DATASET
+        (
+            [39.27, 38.0, 36.06, 37.75, 37.4, 35.5, 36.25, 35.41, 35.76, 34.56,
+             35.05, 33.5, 34.53, 34.11, 33.55, 33.74, 25.2, 24.4, 23.87, 22.73,
+             22.92, 22.33, 23.72, 21.95, 21.53, 21.01, 21.3, 21.16, 21.75, 22.51,
+             22.15, 22.75, 22.9, 21.94, 21.75, 21.5, 21.84, 21.77, 22.44, 23.15, 22.6,
+             22.4, 23.1, 23.0, 23.94, 23.63, 23.46, 23.75, 23.75, 23.75, 23.0, 23.27, 23.82,
+             23.56, 23.5, 22.9, 22.74, 22.13, 22.13, 22.21, 22.35, 22.35, 22.35,
+             22.15, 20.86, 20.7, 20.6, 21.02, 21.0, 21.76, 20.99, 22.0, 22.04, 22.45,
+             22.61, 23.15, 23.41, 21.39, 21.25, 21.09, 18.89, 17.71, 17.13, 17.12,
+             16.99, 16.56, 16.5, 15.71, 15.75, 17.13, 16.6, 16.53, 16.87, 18.09, 18.1,
+             17.83, 18.25, 17.9, 17.91, 18.11],
+            {DECIMAL9_2 n}
+        );
+
+    SHARED BenfordResults1 := DataPatterns.Benford(TestData, 'n', digit := 1);
+    SHARED TestBenford1 := [
+        ASSERT(BenfordResults1[1].zero = (DECIMAL4_1)-1),
+        ASSERT(BenfordResults1[1].one = (DECIMAL4_1)30.1),
+        ASSERT(BenfordResults1[1].two = (DECIMAL4_1)17.6),
+        ASSERT(BenfordResults1[1].three = (DECIMAL4_1)12.5),
+        ASSERT(BenfordResults1[1].four = (DECIMAL4_1)9.7),
+        ASSERT(BenfordResults1[1].five = (DECIMAL4_1)7.9),
+        ASSERT(BenfordResults1[1].six = (DECIMAL4_1)6.7),
+        ASSERT(BenfordResults1[1].seven = (DECIMAL4_1)5.8),
+        ASSERT(BenfordResults1[1].eight = (DECIMAL4_1)5.1),
+        ASSERT(BenfordResults1[1].nine = (DECIMAL4_1)4.6),
+        ASSERT(BenfordResults1[1].chi_squared = (DECIMAL7_3)20.09),
+        ASSERT(BenfordResults1[1].num_values = 100),
+        ASSERT(BenfordResults1[2].zero = (DECIMAL4_1)-1),
+        ASSERT(BenfordResults1[2].one = (DECIMAL4_1)20),
+        ASSERT(BenfordResults1[2].two = (DECIMAL4_1)64),
+        ASSERT(BenfordResults1[2].three = (DECIMAL4_1)16),
+        ASSERT(BenfordResults1[2].four = (DECIMAL4_1)0),
+        ASSERT(BenfordResults1[2].five = (DECIMAL4_1)0),
+        ASSERT(BenfordResults1[2].six = (DECIMAL4_1)0),
+        ASSERT(BenfordResults1[2].seven = (DECIMAL4_1)0),
+        ASSERT(BenfordResults1[2].eight = (DECIMAL4_1)0),
+        ASSERT(BenfordResults1[2].nine = (DECIMAL4_1)0),
+        ASSERT(BenfordResults1[2].chi_squared = (DECIMAL7_3)166.496),
+        ASSERT(BenfordResults1[2].num_values = 100)
+    ];
+
+    SHARED BenfordResults2 := DataPatterns.Benford(TestData, digit := 2);
+    SHARED TestBenford2 := [
+        ASSERT(BenfordResults2[1].zero = (DECIMAL4_1)12),
```

Comment:
FYI You can use D as a suffix on a number to make a decimal constant.  E.g. 11.4D.

Reply:
Much cleaner; changed.